create_sarek_samplesheet.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+
+fqdir = sys.argv[1]
+outfile = sys.argv[2]
+
+sampledict = {}
+fqpattern = r'^(.*)_(S\d+)_L(\d+)_(R\d)_001.*$'
+fcpattern = r'^.*_[AB]?([^_]+)$'
+
+
+for root, dirs, fqlinks in os.walk(fqdir):
+    for link in fqlinks:
+        fqpath = os.readlink(os.path.join(root, link))
+        fqbasenm = re.match(fqpattern, link)
+
+        if fqbasenm:
+            sample = fqbasenm.group(1)
+            ssheet_idx = fqbasenm.group(2)
+            laneno = int(fqbasenm.group(3))
+            readnr = fqbasenm.group(4)
+
+        else:
+            continue
+
+        m = re.match(fcpattern, os.path.basename(os.path.dirname(fqpath)))
+        fcid = m.group(1) if m else "NA"
+
+        readgrp = f"{fcid}.{laneno}.{ssheet_idx}"  # PU, Plattform unit. Will be used as read tag in BAM-files.
+
+        if readgrp not in sampledict:
+            sampledict[readgrp] = {"sample": sample, "R1": "", "R2": ""}
+        sampledict[readgrp][readnr] = fqpath
+
+with open(outfile, 'w') as fout:
```

Comment:
You could consider using the csv library here, but there are not many benefits with it in this case imo, so I think it is fine to leave as it is.

Reply:
I looked at it, but I did not see any real benefits either in this case